Source/WebCore/platform/TrackInfo.h

```diff
@@ -133,103 +143,125 @@ struct TrackInfo : public ThreadSafeRefCounted<TrackInfo> {
     const TrackType m_type { TrackType::Unknown };
 };
 
-struct VideoInfo : public TrackInfo {
-    static Ref<VideoInfo> create() { return adoptRef(*new VideoInfo()); }
+struct EncryptionData {
+    std::optional<TrackInfo::EncryptionData> encryptionData;
+    std::optional<FourCC> encryptionOriginalFormat;
+    Vector<TrackInfo::EncryptionInitData> encryptionInitDatas;
+};
 
-    FloatSize size;
+struct VideoInfoData {
```

Comment:
I would tend to have VideoInfoData = std::pair<TrackInfoData, VideoSpecificInfoData> or a simple structure if is easier;
Then VideoInfo would have its `VideoSpecificInfoData m_data` member and you would not need to have these long constructors and long `to` methods.

Reply:
it would make the construction a tad more verbose with extra level of `{ }` ; I could just VideoInfo be a single member VideoInfoData ...